update-dependencies.py

```diff
@@ -0,0 +1,176 @@
+from os.path import join
+from json import loads
+from typing import Union
+from urllib.request import urlopen
+from hashlib import sha256 as sha256sum
+
+DEPENDENCIES_PATH = join("bazelrio", "deps.bzl")
+WPILIB_VERSION = "2021.3.1"
+NI_VERSION = "2020.9.2"
+REV_SPARKMAX_VERSION = "1.5.4"
+REV_COLORSENSOR_VERSION = "1.2.0"
+CTRE_PHOENIX_VERSION = "5.19.4"
+KAUAILABS_NAVX_VERSION = "4.0.425"
```

Comment:
I wonder if it would make sense to scrape everything under a Maven repo rather than specify versions here. It looks like URLs like https://frcmaven.wpi.edu/api/storage/wpilib-mvn-release/ are valid and return a list of children. Maybe a future feature to get access to older versions.

Reply:
The `{vendor}-mvn-remote` repos don't seem to be reliably populated though... maybe something to take up with the WPILib team... Another thought that's crossed my mind is to run our own Artifactory to reliably do this remote repo caching.

Reply:
Do you want to scrape everything in the sense of "get the latest versions" or rather "download literally everything on the repository"?

Reply:
I was thinking of generating http_archive calls for everything, but not necessarily cc_library defs. If other people want to write them for older versions they'll have an easier time, and we'll have an easier time keeping old versions around as we port forward in the future.

Reply:
I believe this could work well with PJ's suggestion. Implementation will be tomorrow probably

Reply:
I might take a crack at running our own Artifactory this weekend.

Reply:
Are you sure Artifactory is the best way to provide the dependencies? I'm all for the idea of hosting them ourselves.

Reply:
If you know of a tool to scape Maven repos and generate SHAs I'm all for it.

Reply:
I think a simple HTTP file hosting website (like apache I think?) could be optimal.
But I am unfamiliar with any possibilities within Artifactory of scraping other repositories.

Reply:
I think a simple http/s file hosting server (similar to apache I think?) will be optimal.
In any case I am unfamiliar with artifactories' scraping abilities